tfac/figures/figure3.py

```diff
@@ -1,23 +1,68 @@
 """
-This creates Figure 3 - Individual Data performance
+This creates Figure 3 - Cytokine plots
 """
 import pickle
 import numpy as np
 import pandas as pd
 import seaborn as sns
+from scipy.stats import pearsonr
 from .figureCommon import subplotLabel, getSetup
+from ..dataImport import form_missing_tensor
 from ..predict import SVC_predict
 
 
 def fig_3_setup():
-    return
+    tensor_slices, cytokines, _, patInfo = form_missing_tensor()
+    test = pd.concat([pd.DataFrame(tensor_slices[0]), pd.DataFrame(tensor_slices[1])])
+    test = test.dropna(axis=1)
+    pears = []
+    for i in range(38):
+        pears.append([cytokines[i], pearsonr(test.iloc[i, :].to_numpy(dtype=float), test.iloc[i + 38, :].to_numpy(dtype=float))[0]])
+    pears = pd.DataFrame(pears).sort_values(1)
+
+    ser = pd.DataFrame(tensor_slices[0], index=cytokines, columns=patInfo.columns).iloc[:, :148].dropna(axis=1).T
+    ser["Outcome"] = patInfo.loc["status"]
+    out0 = ser[ser["Outcome"] == 0]
+    out1 = ser[ser["Outcome"] == 1]
+    ser = ser.append(pd.DataFrame(np.abs(out0.median() - out1.median()), columns=["diff"]).T).sort_values(by="diff", axis=1).drop("diff")
+    ser = pd.melt(ser, id_vars=["Outcome"])
+
+    plas = pd.DataFrame(tensor_slices[1], index=cytokines, columns=patInfo.columns).iloc[:, :148].dropna(axis=1).T
+    plas["Outcome"] = patInfo.loc["status"]
+    out0 = plas[plas["Outcome"] == 0]
+    out1 = plas[plas["Outcome"] == 1]
+    plas = plas.append(pd.DataFrame(np.abs(out0.median() - out1.median()), columns=["diff"]).T).sort_values(by="diff", axis=1).drop("diff")
+    plas = pd.melt(plas, id_vars=["Outcome"])
+
+    return pears, ser, plas
 
 
 def makeFigure():
     """ Get a list of the axis objects and create a figure. """
     # Get list of axis objects
-    ax, f = getSetup((8, 8), (1, 1))
-    # Add subplot labels
-    subplotLabel(ax)
+    ax, f = getSetup((15, 18), (3, 1))
```

Comment:
This size should not be larger than a width of 8 in and a height of 6 in.

Reply:
To clarify, the largest allowable size is 8x6? Is 6x8 okay?

Reply:
A height of 6 in still allows some body text on a page. The other way around would require a full page figure.

Reply:
Understood, I thought it was scaled down regardless of the size. I'll make sure to adjust that.